kolibri/core/analytics/management/commands/profile.py

```diff
@@ -0,0 +1,78 @@
+import logging
+import time
+from datetime import timedelta
+
+import psutil
+from django.contrib.sessions.models import Session
+from django.core.management.base import BaseCommand
+from django.db import connection
+from django.db.utils import OperationalError
+from django.utils import timezone
+
+from kolibri.core.logger.models import UserSessionLog
+
+logger = logging.getLogger('profiler')
+
+
+class Command(BaseCommand):
+    help = "Logs performance/profiling info in the server running Kolibri"
+
+    def add_arguments(self, parser):
+        parser.add_argument(
+            '--interval', action='store', dest='interval', default=0, type=int,
+            help='Specifies the interval (in seconds) to run the process continuosly. If 0, no repetition will happen',
```

Comment:
It would be nice with some idea of what the desired interval is? I wouldn't know whether to put logs every second or every minute. Is the use case something like:

1) User reports slow server
2) We ask for a performance log, like `kolibri manage profile --interval=X`
3) User uploads performance log

Reply:
Yes, the intent was more users would run it only once.
Maybe we should change it to be running for 30 or 60 seconds by default. Specially thinking of the request metrics, they need requests to be done for some time.

Reply:
What's the lower bound? I mean, at some point the profiling would turn into a DOS attack? :)

30 seconds sounds okay, that way an impatient user would maybe run the profiler for 5-10 minutes and then there would still be enough data?

How much time would we want the profiling to be running for? Say, it's a classroom and there are 30 client devices, so it'll have some fluctuation...

Reply:
No, it will never be a DOS attack, it can affects performance because it writes in a file in the disk, however the impact should be minimal. It's not something making requests to the server, but measuring how long, how much cpu and how much memory the actual requests take.
Of course I would not have it active by default, but having it running for one hour or more in a classroom where we want to detect problems should not suppose an added problem, it should not be noticeable for the user.